From 15dd69296c5607aa718feb071d0508c0a5d0ed9f Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Mon, 1 Jul 2019 10:38:49 +0800
Subject: [PATCH] Tweak label name

In yocto, the product name is too long, which enlarge
the gui window, short them.

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>

Rebase for anaconda 34.

Signed-off-by: Kai Kang <kai.kang@windriver.com>

Rebase for anaconda 37 on 20221020.

Signed-off-by: Kai Kang <kai.kang@windriver.com>

Rebase for anaconda 38 on 20231107.

Signed-off-by: Kai Kang <kai.kang@windriver.com>

Rebase for anaconda 39 on 20240304.

Signed-off-by: Kai Kang <kai.kang@windriver.com>

Rebase for anaconda 42 on 20250328.

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
 pyanaconda/ui/gui/spokes/welcome.py | 4 ++--
 pyanaconda/ui/helpers.py            | 9 ++++++++-
 2 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/pyanaconda/ui/gui/spokes/welcome.py b/pyanaconda/ui/gui/spokes/welcome.py
index c785f28912..388884cfb0 100644
--- a/pyanaconda/ui/gui/spokes/welcome.py
+++ b/pyanaconda/ui/gui/spokes/welcome.py
@@ -37,7 +37,7 @@ from pyanaconda.core.constants import (
 from pyanaconda.core.i18n import _
 from pyanaconda.core.product import (
     get_product_is_final_release,
-    get_product_name,
+    get_product_short_name,
     get_product_version,
 )
 from pyanaconda.core.util import ipmi_abort
@@ -212,7 +212,7 @@ class WelcomeLanguageSpoke(StandaloneSpoke, LangLocaleHandler):
         welcomeLabel = self.builder.get_object("welcomeLabel")
 
         welcomeLabel.set_text(_("WELCOME TO %(name)s %(version)s.") %
-                {"name" : get_product_name().upper(), "version" : get_product_version()})
+                {"name" : get_product_short_name().upper(), "version" : get_product_version()})
 
         # Retranslate the language (filtering) entry's placeholder text
         languageEntry = self.builder.get_object("languageEntry")
diff --git a/pyanaconda/ui/helpers.py b/pyanaconda/ui/helpers.py
index 35adce9cf2..e52e41cd8e 100644
--- a/pyanaconda/ui/helpers.py
+++ b/pyanaconda/ui/helpers.py
@@ -61,12 +61,19 @@ from pyanaconda.core.constants import DRACUT_REPO_DIR
 from pyanaconda.core.i18n import _
 from pyanaconda.core.path import join_paths
 from pyanaconda.core.payload import create_hdd_url, create_nfs_url
-from pyanaconda.core.product import get_product_name, get_product_version
+from pyanaconda.core.product import get_product_name, get_product_short_name, get_product_version
 from pyanaconda.modules.common.structures.payload import RepoConfigurationData
 from pyanaconda.ui.lib.payload import create_source, set_source, tear_down_sources
 
 
 def get_distribution_text():
+    productName = get_product_name()
+    if '(' in productName and ')' in productName:
+        return _("%(productName)s %(productVersion)s INSTALLATION") % {
+            "productName": get_product_short_name().upper(),
+            "productVersion": get_product_version().upper()
+        }
+
     return _("%(productName)s %(productVersion)s INSTALLATION") % {
         "productName": get_product_name().upper(),
         "productVersion": get_product_version().upper()
-- 
2.7.4
